### PR TITLE
Adobe journey optimizer - push messaging guide doc

### PR DESCRIPTION
**Pull request by anshikagarwal-git**

### CODE REVIEW

**Comment by anshikagarwal-git** [comment]
@calebk1 - This documentation is being added for a holistic picture of Push Messaging flow using Adobe Journey optimizer. Please review whenever you find time. Thanks!
Also tagging @chetanprasad0 and @archanaadobe for review.

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Mobile SDK has now integrated with [Project Griffon](https://aep-sdks.gitbook.io/docs/beta/project-griffon), which is a new, innovative beta product from Adobe Experience Cloud to help you inspect, proof, simulate, and validate how you collect data or serve experiences in your mobile app. With respect to push notifications, this integration helps validate various configurations across components involved in the data flow, by running a [test end-to-end push notification flow](https://aep-sdks.gitbook.io/docs/resources/troubleshooting-guides/troubleshooting-push). It shows errors encountered and recommendations to solve misconfigurations.
+
+#### Adobe Launch
+
+[Adobe Launch or Data Collection](https://experienceleague.adobe.com/docs/launch/using/home.html) is the next generation of mobile SDK management capabilities which enables data flow from mobile SDK to Experience Platform. It provides capabilities to register extensions, create rules and data elements to send data from your mobile app to the Adobe experience cloud solutions. With respect to the push notifications data flow, the primary configurations required in Launch are :
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:36]: "... enables data flow from Mobile SDK to Experience Platform."

Comment (calebk1) [reply beta/adobe-journey-optimizer/push-messaging-guide.md:36]: "... to register extensions, create rules, and create data elements to send data from your mobile app to Adobe Experience Cloud solutions."

Comment (calebk1) [reply beta/adobe-journey-optimizer/push-messaging-guide.md:36]: "For the push notifications data flow, the primary configurations required in Launch are:"

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+- Creating a datastream to configure the profile and experience event datasets against which the data flows into experience platform.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:38]: "... the Profile and Experience Event datasets... into Experience Platform."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Adobe Launch
+
+[Adobe Launch or Data Collection](https://experienceleague.adobe.com/docs/launch/using/home.html) is the next generation of mobile SDK management capabilities which enables data flow from mobile SDK to Experience Platform. It provides capabilities to register extensions, create rules and data elements to send data from your mobile app to the Adobe experience cloud solutions. With respect to the push notifications data flow, the primary configurations required in Launch are :
+
+- Creating a datastream to configure the profile and experience event datasets against which the data flows into experience platform. 
+- Creating a client-side mobile property and adding extensions. Mobile SDK closely integrates with these extensions to provide a seamless data collection experience.
+- Registering the mobile app bundle identifier and app credentials that would help to uniquely identify and validate sanity of the app at the time of delivering push notification. 
+
+You will find detailed steps for these configurations [here](https://aep-sdks.gitbook.io/docs/beta/adobe-journey-optimizer).
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:42]: "You can find detailed steps..."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Real-time Customer Profile
+
+[Real-time Customer Profile](https://experienceleague.adobe.com/docs/experience-platform/profile/home.html) is the component in [Experience Platform](https://experienceleague.adobe.com/docs/experience-platform.html) that allows you to see a holistic view of each individual customer by combining data from multiple channels, including web, mobile, CRM, and third party. Profile allows you to consolidate your customer data into a unified view offering an actionable, timestamped account of every customer interaction. 
+
+Static data that identifies the user of the mobile app such as a push token, is stored against the user's profile as record data while the interactions the user does with push notifications are tracked as time-series events data.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:48]: "... the mobile app, such as a push token, is stored against the user's profile as record data, while the user's interactions with push notifications are tracked as time-series event data."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+[Real-time Customer Profile](https://experienceleague.adobe.com/docs/experience-platform/profile/home.html) is the component in [Experience Platform](https://experienceleague.adobe.com/docs/experience-platform.html) that allows you to see a holistic view of each individual customer by combining data from multiple channels, including web, mobile, CRM, and third party. Profile allows you to consolidate your customer data into a unified view offering an actionable, timestamped account of every customer interaction. 
+
+Static data that identifies the user of the mobile app such as a push token, is stored against the user's profile as record data while the interactions the user does with push notifications are tracked as time-series events data. 
+
+#### Adobe Journey Optimizer
+
+Once your mobile app integrations with above mentioned components are in place and your customer profiles are available as Real-time customer profiles, you can take advantage of powerful audience segmentation capabilities in [Adobe Journey Optimizer](https://experienceleague.adobe.com/docs/journey-optimizer/using/ajo-home.html) to ensure the optimal experience for each person.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:52]: "Once your mobile app integrates with the previously mentioned components and your customer profiles are available as Real-Time Customer Profiles, you can..."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+
+
+### Push Data Flow Steps
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:56]: Same comment about title vs. sentence casing.

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+Refer to the data flow diagram to identify what each of these bullets represent :
+
+1. The customer develops a mobile app on Android or iOS that they would release to their users. In order to use the push capability provided by push providers i.e APNS by Apple and FCM by Google, the mobile app registers itself and enables the push capability.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:60]: "In order to use the push capability provided by the providers (APNS for Apple and FCM for Google), the mobile app..."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+#### Adobe Journey Optimizer
+
+Once your mobile app integrations with above mentioned components are in place and your customer profiles are available as Real-time customer profiles, you can take advantage of powerful audience segmentation capabilities in [Adobe Journey Optimizer](https://experienceleague.adobe.com/docs/journey-optimizer/using/ajo-home.html) to ensure the optimal experience for each person.
+
+
+
+### Push Data Flow Steps
+
+Refer to the data flow diagram to identify what each of these bullets represent :
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:58]: To identify what each of the numbers represent, please refer to the data flow diagram:

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+1. The customer develops a mobile app on Android or iOS that they would release to their users. In order to use the push capability provided by push providers i.e APNS by Apple and FCM by Google, the mobile app registers itself and enables the push capability.
+2. The push providers generate and send a push token to the mobile app. A Push Token is an identifier that senders use to target specific devices with a push notification.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:61]: Does push token need to be capitalized?

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+3. The mobile app is integrated with Adobe Mobile SDK which exposes various extensions and APIs. The Messaging extension exposes an API to register the push token into Adobe Experience Platform against customer's profile.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:62]: "... integrated with Mobile SDK, which exposes...".

Also, does messaging need to be capitalized?

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+   Once the mobile app is ready, it is configured in Adobe Launch > App Configurations along with the credentials.
+4. The marketer now authors a push notification in Adobe Journey Optimizer > Messages against the mobile app registered. 
+5. The marketer orchestrates a customer journey defining the flow of events and actions. To send a push notification at a stage of the journey, the marketer adds an action of type 'Message' and associates it with the message authored in the previous step. 
+6. Whenever a customer profile qualifies to receive the push notification for any reason say on trigger of an event or on qualification of a segment, the message is personalized against the profile, if applicable.
+7. The personalized push message is sent for further processing to the Push delivery service.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:67]: Does push need to be capitalized here?

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+8. The push delivery service validates the credentials of the app associated with the message.
+9. The message is sent to the push providers for delivery to the mobile app against the specific push token and credentials.
+10. The push providers send a feedback suggesting whether the message was successfully delivered to the provider or not. In case not, the relevant error message is part of the feedback. This feedback is sent to Adobe reporting for the customer to view in their Journey Live and Global Reports.
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:70]: "If the message was not successfully delivered, the relevant error message is sent as part of the feedback."

File: beta/adobe-journey-optimizer/push-messaging-guide.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+11. In the meantime, the push providers asynchronously deliver the successful push notification to the mobile app.
+12. As the customer interacts with the notification, the impressions such as click/open can then be tracked as Experience Events. The Messaging extension exposes APIs to send tracking events into Adobe Experience Platform against customer's profile.
+
+
+
+## Understanding Push User Flow
```
Comment (calebk1) [inline beta/adobe-journey-optimizer/push-messaging-guide.md:76]: Same comment about sentence vs. title case.

**Review by calebk1 — CHANGES_REQUESTED** [review]
Leaving some feedback.

My biggest comments mostly are about the large gaps between sections (it really only needs to be one line) as well as title casing the headers, when they should be sentence cased, as per the Adobe style guidelines.

**Comment by anshikagarwal-git** [comment]
We are not going ahead with this PR merge. The document has been published as part of AJO docs here :
https://experienceleague.adobe.com/docs/journey-optimizer/using/get-started/configuration/push-config/push-gs.html?lang=en

**Pull request closed without merging**